pkg/workflows/sdk/workflow_spec.go

```diff
@@ -35,3 +152,57 @@ func (w *WorkflowSpec) Steps() []StepDefinition {
 	s = append(s, w.Targets...)
 	return s
 }
+
+func (w *WorkflowSpec) FormatChart() (string, error) {
+	var sb strings.Builder
+	steps := slices.Clone(w.Triggers)
+	steps = append(steps, w.Steps()...)
+	slices.SortFunc(steps, func(a, b StepDefinition) int {
+		return cmp.Or(
+			a.CapabilityType.Compare(b.CapabilityType),
+			cmp.Compare(a.Ref, b.Ref),
+			cmp.Compare(a.ID, b.ID),
+		)
+	})
+	err := tmpl.Execute(&sb, steps)
+	if err != nil {
+		return "", err
+	}
+	return sb.String(), nil
+}
+
+var tmpl = template.Must(template.New("").Funcs(map[string]any{
+	"replace":  strings.ReplaceAll,
+	"parseRef": parseRef,
+}).Parse(`flowchart
+{{ range $i, $step := . }}
+	{{ $ref := .Ref -}}
+	{{ $id := replace .ID "@" "[at]" -}}
+	{{ $name := printf "%s<br><i>(%s)</i>" .CapabilityType $id -}}
+	{{ if .Ref -}}
+		{{ $name = printf "<b>%s</b><br>%s" .Ref $name -}}
+	{{ else -}}
+		{{ $ref = printf "%s%d" "unnamed" $i -}}
+	{{ end -}}
+	{{ if eq .CapabilityType "trigger" -}}
+	{{ $ref }}[\"{{$name}}"/]
+	{{ else if eq .CapabilityType "consensus" -}}
+	{{ $ref }}[["{{$name}}"]]
+	{{ else if eq .CapabilityType "target" -}}
+	{{ $ref }}[/"{{$name}}"\]
+	{{ else -}}
+	{{ $ref }}["{{$name}}"]
+	{{ end -}}
+	{{ if .Inputs.OutputRef -}}
+	{{ .Inputs.OutputRef }} --> {{ $step.Ref }}
+	{{ else -}}
+		{{ range $out := .Inputs.Outputs -}}
+			{{ if $out.Name -}}
+	{{ $out.Ref }} -- {{ $out.Name }} --> {{ $ref}} 
```

Comment:
I think this is where we indicate that we're doing outputs.Value, right? If so, can we also represent the fact that we also shaping the incoming inputs?

Eg. for consensus we actually do:
{"observations": []any{"outputs.Value"}}

Reply:
I was worried about including everything, and it's not obvious to me where to draw the line :shrug: 
We also have other options for presenting the data, like triggering actions on click or tool-tips on hover.
Or something way different, like subgraphs for each step, with multiple internal nodes:
```mermaid
flowchart
	subgraph event-trigger
		trigger-->event-outputs[outputs]
	end
	subgraph compute-a-step[compute a]
		compute-a-inputs[TriggerOutputs] --> compute-a[Execute] --> compute-a-outputs[Value]
	end
	subgraph compute-b-step[compute b]
		compute-b-inputs[TriggerOutputs] --> compute-b[Execute] --> compute-b-outputs[Value]
	end
	subgraph report-step[report]
		report[Build Report]
		report-inputs-compute-a[Value] --> report
		report-inputs-compute-b[Value] --> report
		report --> report-outputs[Report]
	end
	subgraph write-step[write]
		write-inputs[Report] --> write
	end
	event-outputs --> compute-a-inputs
	event-outputs --> compute-b-inputs
	compute-a-outputs --> report-inputs-compute-a
	compute-b-outputs --> report-inputs-compute-b
	report-outputs --> write-inputs
```
This level of granularity would allow us to express more visually, and utilize tool tips for more information, since they work on nodes but not edges.